src/enclave_proc/resource_manager.rs

```diff
@@ -1164,3 +1176,69 @@ pub fn between_packets_delay() -> Option<Duration> {
 
     None
 }
+
+/// Helper function which contains heuristic for enclave build timeout calculation
+///
+/// If the `eif_size` parameter is greater than 0 then the timeout is based on the eif size
+/// otherwise the timeout is based on the allocated memory
+/// # Arguments
+///
+/// * `eif_size` - The EIF size in bytes
+/// * `allocated_memory_mib` - The allocated memory in MiB
+///
+/// # Examples
+///
+/// ```
+/// // Returns the timeout based on the 4GiB allocated memory
+/// let timeout = calculate_necessary_timeout(0, 1024 * 4);
+///
+/// // Returns the timeout based on the 8GiB EIF size
+/// let timeout = calculate_necessary_timeout(8 * GiB, 0);
+/// ```
+fn calculate_necessary_timeout(eif_size: u64, allocated_memory_mib: u64) -> c_int {
+    let poll_timeout: c_int;
+
+    if eif_size > 0 {
+        // in case we have a valid eif_size give 1.5 minutes for each 6GiB
+        poll_timeout = ((1 + (eif_size - 1) / (6 * GiB)) as i32).saturating_mul(TIMEOUT_MINUTE_MS);
+    } else {
+        // Update the poll timeout to be 1 minute per 100 GiB of enclave memory.
```

Comment:
Why 1 min per 100 GB? was this measured somehow?

We recommend to customers that the enclave memory is 4 times the EIF size, minimum. This means that the EIF size here could be 25 GB.

Reply:
This was the previous implementation and it did not take into account the EIF size.

Reply:
I eliminated the legacy way of calculating the timeout.